src/java/org/apache/cassandra/db/lifecycle/Tracker.java

```diff
@@ -389,29 +390,24 @@ else if (cfstore != null && !cfstore.isValid())
             accumulate = Throwables.merge(accumulate, t);
         }
 
-        logger.debug("Sstables for {} dropped with operation {}: {}", 
-                     metadata.name, operationType, accumulate == null ? "null" : accumulate.getMessage());
+        logger.debug("Sstables for {} {} with operation {}: {}",
+                     metadata.name, operationType.localOnly ? "unloadded" : "dropped", operationType, accumulate == null ? "null" : accumulate.getMessage());
         return accumulate;
     }
 
     /**
-     * Unload all sstables from current tracker without deleting files
+     * Unload all sstables from current tracker without deleting remote files
      */
+
     public void unloadSSTables()
     {
         maybeFail(unloadSSTables(null));
     }
 
     public Throwable unloadSSTables(@Nullable Throwable accumulate)
     {
-        Pair<View, View> result = apply(view -> {
-            Set<SSTableReader> toUnload = copyOf(filter(view.sstables, notIn(view.compacting)));
-            return updateLiveSet(toUnload, emptySet()).apply(view);
-        });
-
-        // compacting sstables will be cleaned up by their transaction in {@link LifecycleTransaction#unmarkCompacting}
-        Set<SSTableReader> toRelease = Sets.difference(result.left.sstables, result.right.sstables);
-        return release(selfRefs(toRelease), accumulate);
+        // SSTABLE_DISCARD uses local transaction which deletes files locally
+        return dropSSTables(Predicates.alwaysTrue(), OperationType.SSTABLE_DISCARD, accumulate);
```

Comment:
I think I understand the logic of the change, but I'm a bit worried by it as it:
- this essentially make "unload" means "delete sstables" and that make the naming pretty confusing imo. I could easily see this leading to confusing and errors, and you kind of need to know the cndb context to understand the reason.
- this only feel like the right thing for CNDB, not usual C* and while I understand we're ok prioritizing making things work for CNDB over making it easier to later upstream to OSS, that still feels a bit ... strong?.

That said, not sure what is a simple and clean way to abstract this. If we'd rather not spend time on this, I guess I'd be fine if we at least:
- rename `unloadSSTables` to `dropSSTablesLocally` to remove any confusion on what it does.
- add some comment in CFS when calling that method to explain that the behavior make sense for CNDB but that for C* proper, this will end up truly dropping data (despite the `dropData` boolean being false.

Reply:
Right, it's not ideal.

An alternative approach would be to add a new API `StorageProvider#invalidateFileSystemCache(Descriptor desc, boolean tidied)` to replace existing `invalidateFileSystemCache(file)` at the end of `GlobalTidy#tidy()`. So CNDB can intercept this call and clear remote file cache for archived components as well as data/partition-index/row-index.

WDYT?

Reply:
pushed here https://github.com/datastax/cassandra/pull/1176/commits/74ac7528f4aceb40341664824615e69d28928f87

Reply:
I like that option!